core/src/main/scala/org/apache/spark/sql/rapids/tool/store/StageModel.scala

```diff
@@ -27,11 +27,34 @@ import org.apache.spark.sql.rapids.tool.annotation.{Calculated, Since, WallClock
  *
  * @param sInfo Snapshot from the stage info loaded from the eventlog
  */
+
 @Since("24.02.3")
-class StageModel private(var sInfo: StageInfo) {
+class StageModel private(sInfo: StageInfo) {
 
-  val sId: Int = sInfo.stageId
-  val attemptId: Int = sInfo.attemptNumber()
+  var stageInfo: StageInfo = _
+  updateInfo(sInfo)
+
+  /**
+   * This method create a new StageInfo object from the incoming StageInfo object.
```

Comment:
Nit:  Creates a new StageInfo object by extracting and copying only the necessary fields from the given StageInfo object.

Reply:
Updated